Swift 5.0 updates and add support for encoding dictionaries with Any values

hamchapman opened the pull request:
You may well only want part of this (i.e. the updates to Swift 5), so I can create a separate PR if needed.

The other change is to support encoding of a map with type `[A: Any]` where `A: CBOREncodable`.

To make handling `Data` easy I added an `import Foundation`, but I know that that might be unpopular!

If you would like to have the encoding for dictionaries with `Any` values but don't want the dependency on Foundation then I can make that conditional with `#if canImport(Foundation)` and only add the `Data` case if that is fulfilled, and in that case I'd also add a case to support `[UInt8]` directly.

The last thing to note with that map encoding is that I made it so that the keys are ordered when the encoding happens. This is mentioned in the RFC but doesn't seem to be universally adhered to. Again, something I'd be happy to hear your thoughts on.

Let me know what you think.

myfreeweb commented on Sources/SwiftCBOR/CBOR.swift:
```diff
@@ -19,23 +19,24 @@ public indirect enum CBOR : Equatable, Hashable,
     case `break`
 
-    public var hashValue : Int {
+    public func hash(into hasher: inout Hasher) {
```
Is this for Swift 5? Does this break Swift 4?

hamchapman replied:
I think the `Hasher` changes were actually a part of Swift 4.2.

If you'd like I can leave the old code in and add a `#if swift(>=5.0)` and include the new code in that, just to be extra safe?

myfreeweb commented on Sources/SwiftCBOR/CBOREncoder.swift:
```diff
@@ -1,3 +1,4 @@
+import Foundation
```
eh, I guess I don't have a reason to avoid Foundation — that decision was made before the cross-platform (Linux) version of Swift was released, I thought it might not include Foundation, but it does.

hamchapman replied:
I've made the import conditional using `#if canImport(Foundation)`

myfreeweb replied:
cool

hamchapman commented:
I've added a small test for the new behaviour I added, as well as fixing a flaky test I found.

Let me know what you think about the conditional code for the hashing behaviour 👍

myfreeweb commented:
LGTM, thanks
